### PR TITLE
Modal + Pokemon Modal Card Components

### PR DESCRIPTION
**Pull request by IlanSirisky**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,10 @@
     "preview": "vite preview"
   },
   "dependencies": {
+    "@emotion/react": "^11.11.4",
+    "@emotion/styled": "^11.11.5",
+    "@mui/icons-material": "^5.15.19",
+    "@mui/material": "^5.15.19",
```
Comment (oranroz11) [inline package.json]: remove ^

File: src/components/Modal/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import xCloseIcon from "../../assets/icons/xCloseIcon.svg";
+import { ModalContent, CloseButton, modalStyles } from "./styles";
+import { Modal as MuiModal, Box } from "@mui/material";
+
+interface ModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  children: React.ReactNode;
+  closeIcon?: string;
+  sx?: object;
```
Comment (oranroz11) [inline src/components/Modal/Modal.tsx]: add type

File: src/components/Modal/styles.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  position: "absolute",
+  top: "50%",
+  left: "50%",
+  transform: "translate(-50%, -50%)",
+};
+
+export const ModalContent = styled(Box)`
+  position: relative;
+  width: 500px;
+  box-shadow: 1px 1px 25px 0px #71787f1a;
```
Comment (oranroz11) [inline src/components/Modal/styles.ts]: use spacing and color spacing

File: src/components/Modal/styles.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+`;
+
+export const CloseButton = styled(IconButton)`
+  position: absolute;
+  top: 10px;
+  right: 10px;
+  z-index: 1;
```
Comment (oranroz11) [inline src/components/Modal/styles.ts]: why do you need z-index?

File: src/features/PokemonModalCard/PokemonModalCard.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  HeadingMediumRegular,
+  SubheadingRegular,
+} from "../../styles/typography";
+
+interface PokemonModalCardProps {
+  title: string;
+  subheadText: string;
+  image: string;
+  description: string;
+  attributes: { label: string; value: string }[];
```
Comment (oranroz11) [inline src/features/PokemonModalCard/PokemonModalCard.tsx]: create type for attributes

File: src/features/PokemonModalCard/PokemonModalCard.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  subheadText,
+  image,
+  description,
+  attributes,
+  style,
+}: PokemonModalCardProps) => {
+  return (
+    <PokemonModalCardContainer style={style}>
+      <HeaderContainer>
+        <StyledHeaderContent>
```
Comment (oranroz11) [inline src/features/PokemonModalCard/PokemonModalCard.tsx]: this both component are wrapping the same thing, so why we have two different components for that?

File: src/features/PokemonModalCard/styles.ts [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+const { Neutrals } = COLORS;
+
+export const PokemonModalCardContainer = styled(flexColumn)`
+  background-color: ${Neutrals.White};
+  width: 100%;
+`;
+
+export const HeaderContainer = styled.div`
+  padding: ${cssSpacings.s12} ${cssSpacings.s24};
+  gap: "2px";
```
Comment (oranroz11) [inline src/features/PokemonModalCard/styles.ts]: use spacing

**Review by oranroz11 — CHANGES_REQUESTED** [review]

File: src/components/Modal/Modal.tsx (hunk 2 above)
Comment (IlanSirisky) [reply src/components/Modal/Modal.tsx]: Didnt understand

**Review by YuvalLeviMoveoHLS — APPROVED** [review]

**Pull request merged**